### PR TITLE
Add upload permissions for configuration-as-code-test-harness

### PR DESCRIPTION
**Pull request by timja**
# Description

Add permissions for new test-harness for configuration-as-code-plugin: https://github.com/jenkinsci/configuration-as-code-plugin/pull/1215

And update @casz to be security contact for snakeyaml component (he's the contact for all other configuration-as-code component's, this one was missed in the update.

# Submitter checklist for changing permissions

### Always

- [x] Add link to plugin/component Git repository in description above

### For a new permissions file only

- [x] Make sure the file is created in `permissions/` directory
- [x] `artifactId` (pom.xml) is used for `name` (permissions YAML file).
- [x] [`groupId` / `artifactId` (pom.xml) are correctly represented in `path` (permissions YAML file)](https://github.com/jenkins-infra/repository-permissions-updater/#managing-permissions)

### CODE REVIEW

File: permissions/component-configuration-as-code-test-harness.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+paths:
+- "io/jenkins/configuration-as-code/test-harness"
+developers:
+- "praqma"
+- "casz"
+- "oleg_nenashev"
+- "timja"
+security:
+  contacts:
+    jira: "casZ"
```
Comment (casz) [inline permissions/component-configuration-as-code-test-harness.yml]:
```suggestion
    jira: "casz"
```

**Comment by timja** [comment]
> [2019-12-07T18:19:52.298Z] [ERROR]     Non-resolvable parent POM for io.jenkins.infra:repository-permissions-updater:1.0-SNAPSHOT: Could not transfer artifact org.jenkins-ci:jenkins:pom:1.36 from/to repo.jenkins-ci.org (https://repo.jenkins-ci.org/public/): Failed to transfer file: https://repo.jenkins-ci.org/public/org/jenkins-ci/jenkins/1.36/jenkins-1.36.pom. Return code is: 502 , ReasonPhrase:Bad Gateway. and 'parent.relativePath' points at wrong local POM @ line 3, column 13 -> [Help 2]

**Comment by slide** [comment]
Needs auth from @ewelinawilkosz for the change to the component yaml.

**Comment by timja** [comment]
> Needs auth from @ewelinawilkosz for the change to the component yaml.

It was authorised here:
https://github.com/jenkins-infra/repository-permissions-updater/pull/1180

One was missed

**Pull request merged**